test/e2e/sagemaker/tests/test_endpoint.py

```diff
@@ -0,0 +1,264 @@
+    assert model_resource is not None
+
+    config1_reference = k8s.CustomResourceReference(
+        CRD_GROUP,
+        CRD_VERSION,
+        CONFIG_RESOURCE_PLURAL,
+        config1_resource_name,
+        namespace="default",
+    )
+    config1_resource = k8s.create_custom_resource(config1_reference, config)
+    config1_resource = k8s.wait_resource_consumed_by_controller(config1_reference)
+    assert config1_resource is not None
+
+    config2_resource_name = random_suffix_name("2-single-variant-endpoint", 32)
+    config["metadata"]["name"] = config["spec"][
+        "endpointConfigName"
+    ] = config2_resource_name
+    logging.debug(config)
+    config2_reference = k8s.CustomResourceReference(
+        CRD_GROUP,
+        CRD_VERSION,
+        CONFIG_RESOURCE_PLURAL,
+        config2_resource_name,
+        namespace="default",
+    )
+    config2_resource = k8s.create_custom_resource(config2_reference, config)
+    config2_resource = k8s.wait_resource_consumed_by_controller(config2_reference)
+    assert config2_resource is not None
+
+    endpoint_reference = k8s.CustomResourceReference(
+        CRD_GROUP,
+        CRD_VERSION,
+        ENDPOINT_RESOURCE_PLURAL,
+        endpoint_resource_name,
+        namespace="default",
+    )
+    endpoint_resource = k8s.create_custom_resource(endpoint_reference, endpoint_spec)
+    endpoint_resource = k8s.wait_resource_consumed_by_controller(endpoint_reference)
+    assert endpoint_resource is not None
+
+    yield (endpoint_reference, endpoint_resource, endpoint_spec, config2_resource_name)
+
+    # Delete the k8s resource if not already deleted by tests
+    try:
+        k8s.delete_custom_resource(model_reference)
+        k8s.delete_custom_resource(config1_reference)
+        k8s.delete_custom_resource(config2_reference)
+        k8s.delete_custom_resource(endpoint_reference)
+    except:
+        pass
+
+
+@service_marker
+@pytest.mark.canary
+class TestEndpoint:
+    status_creating: str = "Creating"
+    status_inservice: str = "InService"
+    status_udpating: str = "Updating"
+
+    def _get_resource_endpoint_arn(self, resource: Dict):
+        assert (
+            "ackResourceMetadata" in resource["status"]
+            and "arn" in resource["status"]["ackResourceMetadata"]
+        )
+        return resource["status"]["ackResourceMetadata"]["arn"]
+
+    def _describe_sagemaker_endpoint(self, sagemaker_client, endpoint_name: str):
+        try:
+            return sagemaker_client.describe_endpoint(EndpointName=endpoint_name)
+        except BaseException:
+            logging.error(
+                f"SageMaker could not find a endpoint with the name {endpoint_name}"
+            )
+            return None
+
+    def _wait_resource_endpoint_status(
+        self,
+        reference: k8s.CustomResourceReference,
+        expected_status: str,
+        wait_periods: int = 25,
+    ):
+        resource_status = None
+        while wait_periods > 0:
+            time.sleep(30)
```

Comment:
`25 * 30 seconds = 12.5 minutes`. That's a long time for something to fail. Do we have an average TTL for endpoints?